packages/core/src/sync-store/postgres/store.ts

```diff
@@ -608,15 +779,27 @@ export class PostgresSyncStore implements SyncStore {
           await tx
             .insertInto("transactionReceipts")
             .values(transactionReceipts)
-            .onConflict((oc) => oc.column("transactionHash").doNothing())
+            .onConflict((oc) =>
+              oc.column("transactionHash").doUpdateSet((eb) => ({
+                blockHash: eb.ref("excluded.blockHash"),
+                blockNumber: eb.ref("excluded.blockNumber"),
+                contractAddress: eb.ref("excluded.contractAddress"),
+                cumulativeGasUsed: eb.ref("excluded.cumulativeGasUsed"),
+                effectiveGasPrice: eb.ref("excluded.effectiveGasPrice"),
+                gasUsed: eb.ref("excluded.gasUsed"),
+                logs: eb.ref("excluded.logs"),
+                logsBloom: eb.ref("excluded.logsBloom"),
+                transactionIndex: eb.ref("excluded.transactionIndex"),
+              })),
+            )
```

Comment:
I believe that transaction receipts were experiencing the same reorg bug that was happening with transactions. Main problem is that transaction.hash is not actually a unique identifier because it doesn't reflect any changes when a reorg happens